packages/node_modules/@ciscospark/internal-plugin-conversation/src/conversation.js

```diff
@@ -803,18 +803,32 @@ const Conversation = SparkPlugin.extend({
    * @returns {Promise}
    */
   _inferConversationUrl(conversation) {
-    if (!conversation.url && conversation.id) {
-      return this.spark.internal.device.getServiceUrl('conversation')
-        .then((url) => {
-          conversation.url = `${url}/conversations/${conversation.id}`;
-          /* istanbul ignore else */
-          if (process.env.NODE_ENV !== 'production') {
-            this.logger.warn('conversation: inferred conversation url from conversation id; please pass whole conversation objects to Conversation methods');
+    if (conversation.id) {
+      return this.spark.internal.feature.getFeature('developer', 'web-ha-messaging')
+        .then((haMessagingEnabled) => {
+          if (haMessagingEnabled) {
+            // recompute conversation URL each time as the host may have changed
```

Comment:
Moved to using the feature plugin rather than accessing the feature value in device directly.